src/lib.rs

```diff
@@ -0,0 +1,153 @@
+use substreams::store;
+use substreams::{log, Hex, proto};
+use substreams::errors::Error;
+use substreams_ethereum::{pb::eth::v2 as eth, Event as EventTrait};
+
+#[substreams::handlers::map]
+fn map_transfers(blk: eth::Block) -> Result<AirTransfers, Error> {
+    let mut air_transfers = vec![];
+    let mut air_transfer_batchs = vec![];
+    let mut air_transfer_singles = vec![];
+    for log in blk.logs(){
+        if let Some(event) = abis::erc20::events::Transfer::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            // log::info!("ERC20 token: {}, {}, {}", Hex(log.log.clone().address), log.receipt.transaction.hash.clone().to_string(), log.ordinal());
+
+            let air_transfer: AirTransfer = AirTransfer{
+                amount: event.value.to_string(),
+                token_id: Hex(log.log.clone().address).to_string(),
+                hash: Hex(log.receipt.transaction.hash.clone()).to_string(),
+                from: Hex(event.from).to_string(),
+                to: Hex(event.to).to_string(),
+                token_type: 0,
+                block_number: blk.clone().number,
+                block_timestamp: blk.clone()
+                .header
+                .as_ref()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap()
+                .seconds
+                .to_string(),
+                log_ordinal: log.ordinal(),
+                ..Default::default()
+            };
+    
+            air_transfers.push(air_transfer);
+        }
+        if let Some(event) = abis::erc721::events::Transfer::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            // log::info!("ERC721 token: {}, {}", log.receipt.transaction.hash.clone().to_string(), log.ordinal());
+            let air_transfer: AirTransfer = AirTransfer{
+                amount: 1.to_string(),
+                token_id: event.token_id.to_string(),
+                hash: Hex(log.receipt.transaction.hash.clone()).to_string(),
+                from: Hex(event.from).to_string(),
+                to: Hex(event.to).to_string(),
+                token_type:2,
+                block_number: blk.clone().number,
+                block_timestamp: blk.clone()
+                .header
+                .as_ref()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap()
+                .seconds
+                .to_string(),
+                log_ordinal: log.ordinal(),
+                ..Default::default()
+            };
+            air_transfers.push(air_transfer);
+        }
+        if let Some(event) = abis::erc1155::events::TransferBatch::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            // log::info!("transfer values: {}", &event.values);
+            let air_transfer_batch: AirErc1155TransferBatch = AirErc1155TransferBatch{
```

Comment:
please rename this to erc1155_transfer_batch

Reply:
Done